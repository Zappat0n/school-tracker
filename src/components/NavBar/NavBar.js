import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './NavBar.scss';

const NavBar = ({ title }) => {
  const toggleMenu = (e) => {
    e.preventDefault();
    const content = document.querySelector('.menu-content');
    if (content.classList.contains('d-none')) {
      content.classList.remove('d-none');
    } else {
      content.classList.add('d-none');
    }
  };

  return (
    <>
      <div className="nav-bar">
        <div className="left nav-col ta-left">
          <i id="menu-toggle" role="menubar" tabIndex={0} className="fas fa-bars" onKeyDown={(e) => toggleMenu(e)} onClick={(e) => toggleMenu(e)} />
        </div>
        <div className="center nav-col ta-center">
          {title}
        </div>
        <div className="right nav-col ta-right">
          <Link to="/user/signin" className="link">
            <i className="fas fa-user" />
          </Link>
        </div>
      </div>
      <div className="menu-content d-none">
        <Link to="/classrooms/" className="link">Classrooms</Link>
        <Link to="/presentation_areas/" className="link">Curriculum</Link>
        <Link to="/presentation_subareas/" className="link">Subareas</Link>
      </div>
    </>
  );
};

NavBar.propTypes = {
  title: PropTypes.string.isRequired,
};

export default NavBar;
