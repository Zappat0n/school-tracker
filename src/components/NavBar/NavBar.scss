@import "../../index.scss";

.nav-col {
  width: 33%;
}

.nav-bar {
  display: flex;
  background: $primary-blue;

  @include big-font;

  color: white;
  padding: 0 1rem;

  button {
    background: $primary-blue;
    border: none;
    outline: none;
    font-size: 1.5rem;
    color: white;
  }

  button:hover,
  #menu-toggle:hover {
    cursor: pointer;
  }

  a {
    color: white;
  }
}

.menu-content {
  width: 100%;
  background: white;

  @include big-font;

  border-bottom: 1px solid $primary-blue;

  a {
    padding: 0 1rem;
    color: $primary-blue;
  }
}
