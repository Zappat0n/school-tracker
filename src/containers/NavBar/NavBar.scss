@import "../../index.scss";

.nav-bar {
  display: flex;
  justify-content: space-between;
  background: $primary-blue;
  font: normal normal bold 1.5rem/3rem 'Free Sans', sans-serif;
  color: white;

  button {
    background: $primary-blue;
    border: none;
    outline:none;
    font-size: 1.5rem;
    color: white;
    padding: 0 1rem;
  }

  button:hover {
    cursor: pointer;
  }
}
