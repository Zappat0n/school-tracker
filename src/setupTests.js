// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import { createStore, combineReducers } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import userReducer from './slices/userSlice';
import classroomTableReducer from './slices/classroomTableSlice';

const rootReducer = combineReducers({
  user: userReducer, classroomTable: classroomTableReducer,
});

const initialState = {
  user: {
    tables: {},
    title: '',
  },
  classroomTable: {
    students: [],
    presentations: [],
    scores: {},
  },
};

const store = createStore(rootReducer, initialState);

const renderComponent = (component) => render(
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

const rendererForComponent = (component) => renderer.create(
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export { renderComponent, rendererForComponent, store };
