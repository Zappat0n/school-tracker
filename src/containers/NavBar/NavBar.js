import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import './NavBar.scss';

const NavBar = () => {
  const title = useSelector((state) => state.user.title);

  const toggleMenu = (e) => {
    e.preventDefault();
    const content = document.querySelector('.menu-content');
    if (content.classList.contains('d-none')) {
      content.classList.remove('d-none');
    } else {
      content.classList.add('d-none');
    }
  };

  return (
    <>
      <div className="nav-bar">
        <div className="left nav-col ta-left">
          <i id="menu-toggle" role="none" className="fas fa-bars" onKeyDown={(e) => toggleMenu(e)} onClick={(e) => toggleMenu(e)} />
        </div>
        <div className="center nav-col ta-center">
          {title || 'Welcome'}
        </div>
        <div className="right nav-col ta-right">
          <Link to="/user/signin" className="link">
            <i className="fas fa-user" />
          </Link>
        </div>
      </div>
      <div className="menu-content d-none">
        <Link to="/classrooms/" className="link">Classrooms</Link>
        <Link to="/presentation_areas/" className="link">Curriculum</Link>
      </div>
    </>
  );
};

export default NavBar;
