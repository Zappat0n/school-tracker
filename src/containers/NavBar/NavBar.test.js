import React from 'react';
import '@testing-library/jest-dom';
import { screen, fireEvent } from '@testing-library/react';
import { renderComponent, rendererForComponent } from '../../setupTests';
import NavBar from './NavBar';

test('Renders Input fields correctly', () => {
  renderComponent(
    <>
      <NavBar />
    </>,
  );

  expect(document.querySelector('.d-none')).toBeInTheDocument();
  fireEvent.click(screen.getByRole('menubar'));
  expect(screen.getByText('Classrooms')).toBeInTheDocument();
  expect(document.querySelector('.d-none')).not.toBeInTheDocument();
});

test('NavBar Match SnapShot', () => {
  const tree = rendererForComponent(
    <>
      <NavBar />
    </>,
  );
  expect(tree).toMatchSnapshot();
});
