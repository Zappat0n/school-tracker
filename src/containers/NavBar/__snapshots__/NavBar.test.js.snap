// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Match SnapShot 1`] = `
Array [
  <div
    className="nav-bar"
  >
    <div
      className="left nav-col ta-left"
    >
      <i
        className="fas fa-bars"
        id="menu-toggle"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="menubar"
        tabIndex={0}
      />
    </div>
    <div
      className="center nav-col ta-center"
    >
      Welcome
    </div>
    <div
      className="right nav-col ta-right"
    >
      <a
        className="link"
        href="/user/signin"
        onClick={[Function]}
      >
        <i
          className="fas fa-user"
        />
      </a>
    </div>
  </div>,
  <div
    className="menu-content d-none"
  >
    <a
      className="link"
      href="/classrooms/"
      onClick={[Function]}
    >
      Classrooms
    </a>
    <a
      className="link"
      href="/presentation_areas/"
      onClick={[Function]}
    >
      Curriculum
    </a>
    <a
      className="link"
      href="/presentation_subareas/"
      onClick={[Function]}
    >
      Subareas
    </a>
    <a
      className="link"
      href="/classrooms/1/scores/"
      onClick={[Function]}
    >
      Scores
    </a>
  </div>,
]
`;
